.github/workflows/test.yaml

```diff
@@ -0,0 +1,16 @@
+name: Tests
+
+on:
+  pull_request:
+
+jobs:
+  unit-test:
+    name: Unit tests
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v3
+      - name: Install tox
+        run: |
+          python -m pip install --upgrade pip
+          pip install tox
+      - run: tox -e unit
```

Comment:
Should this use the `operator-workflows`?

Reply:
As noted, out of scope for this PR

Reply:
Out of curiosity, why is this out of scope?

Reply:
I'll spend a lot of time adapting for it to work accordingly to the tests of operator-workflows before changing large portions of it for the rework to sidecar. I think that it's best that I handle both in the sidecar rework (next PR)

Reply:
Iirc the testing workflow just executes the default tox target

Reply:
https://github.com/canonical/operator-workflows/blob/main/.github/workflows/test.yaml

Reply:
There is a lot of issues right now with mypy types, inclusive naming issues, pydocstyle complaining and I even have nightmares of imagining what pylint will say

Reply:
Understood. Note that if you don't define those as default tox targets, they won't be executed.